### PR TITLE
accept start transaction statements

### LINKED ISSUE #11214: CrateDB does not support PostgreSQL's START TRANSACTION
**CrateDB version**:
4.5.0

**Environment description**:
Docker v20.10.5, the issue is reproducible without Docker as well.

**Problem description**:
CrateDB does not support the `START TRANSACTION` statement ([PostgreSQL documentation](https://www.postgresql.org/docs/13/sql-start-transaction.html)). `START TRANSACTION` is an alias of `BEGIN` according to the documentation. While CrateDB does support `BEGIN` as a no-op, `START TRANSACTION` is unsupported and throws an error.
This issue surfaces when trying to connect from a PostgreSQL instance (version 12.6) to CrateDB via foreign data wrapper.

**Steps to reproduce**:
On CrateDB, create a simple empty test table:
```
CREATE TABLE IF NOT EXISTS "docs"."test" (
   "a" INTEGER,
   "b" TEXT
);
```

On PostgreSQL, connect to CrateDB:
```
$ psql -U postgres -d demo -h 127.0.0.1
demo=# CREATE SERVER crate FOREIGN DATA WRAPPER postgres_fdw OPTIONS (dbname 'doc', host '<CrateDB host>', port '5432');
CREATE SERVER
demo=# CREATE USER MAPPING FOR postgres SERVER crate OPTIONS (user '<CrateDB user>', password '<CrateDB password>');
CREATE USER MAPPING
demo=# CREATE FOREIGN TABLE test_fdw(a INTEGER, b TEXT) SERVER crate OPTIONS (schema_name 'docs', table_name 'test');
CREATE FOREIGN TABLE
demo=# SELECT * FROM test_fdw;
ERROR:  line 1:1: mismatched input 'START' expecting {'SELECT', 'DEALLOCATE', 'CREATE', 'ALTER', 'KILL', 'BEGIN', 'COMMIT', 'ANALYZE', 'DISCARD', 'EXPLAIN', 'SHOW', 'OPTIMIZE', 'REFRESH', 'RESTORE', 'DROP', 'INSERT', 'VALUES', 'DELETE', 'UPDATE', 'SET', 'RESET', 'COPY', 'GRANT', 'DENY', 'REVOKE'}
CONTEXT:  remote SQL command: START TRANSACTION ISOLATION LEVEL REPEATABLE READ
```

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB

accepting ``START TRANSACTION`` statements for Postgres compatibility, but as CrateDB does not support transactions, this command has no effect.

**originally:**
```SQL
cr> START TRANSACTION ISOLATION LEVEL REPEATABLE READ;
SQLParseException[line 1:1: mismatched input 'START' expecting {'SELECT', 'DEALLOCATE', 'CREATE', 'ALTER', 'KILL', 'BEGIN', 'COMMIT', 'ANALYZE', 'DISCARD', 'EXPLAIN', 'SHOW', 'OPTIMIZE', 'REFRESH', 'RESTORE', 'DROP', 'INSERT', 'VALUES', 'DELETE', 'UPDATE', 'SET', 'RESET', 'COPY', 'GRANT', 'DENY', 'REVOKE', START}]
```

closes #11214 

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by jeeminso** [comment]
**now:**
```SQL
cr> START TRANSACTION ISOLATION LEVEL REPEATABLE READ;
START OK, 0 rows affected  (0.013 sec)
```

File: docs/interfaces/postgres.rst
Comment (jeeminso) [inline docs/interfaces/postgres.rst:271]: statement > statements right?

File: docs/interfaces/postgres.rst
Comment (mfussenegger) [reply docs/interfaces/postgres.rst:260]: yep

File: docs/interfaces/postgres.rst
Comment (mfussenegger) [reply docs/interfaces/postgres.rst:271]: yes :+1:

**Review by mfussenegger — APPROVED** [review]

File: docs/interfaces/postgres.rst
Comment (jeeminso) [reply docs/interfaces/postgres.rst]: added

File: docs/interfaces/postgres.rst
Comment (jeeminso) [reply docs/interfaces/postgres.rst]: added thanks!

File: docs/appendices/release-notes/unreleased.rst
Comment (seut) [inline docs/appendices/release-notes/unreleased.rst:76]:
```suggestion
- Added support for the ``START TRANSACTION`` statement for Postgres compatibility, but
  as CrateDB does not support transactions, this command has no effect.
```

Comment (jeeminso) [reply docs/appendices/release-notes/unreleased.rst:76]: Thanks, will make the change. Initially I have the same wording but thought it interfered with the second part of the sentence where it then says Crate does not support.

File: docs/sql/statements/start-transaction.rst
Comment (norosa) [inline docs/sql/statements/start-transaction.rst:2]:
```suggestion
.. highlight:: psql

.. sql-start-transaction:
```

this is the new recommended style for RST labels. not a lot of files use this yet though but it's good to get it in at the beginning for new docs

File: docs/sql/statements/start-transaction.rst
Comment (norosa) [inline docs/sql/statements/start-transaction.rst:8]:
```suggestion
The ``START TRANSACTION` statement starts a transaction block.
```

want to expand this a little bit. I think some of the other SQL syntax documents could do with similar improvements, but it's good to get this in early for new docs

I would also like to add a link to a page that explains what transactions are and how you can use them. I don't think we have a doc like that yet. I am guessing that is because the whole feature set is still in development

do you know if there are any existing plans to add a new doc to the [Quering](https://crate.io/docs/crate/reference/en/4.5/general/dql/index.html) section in the docs?

File: docs/sql/statements/start-transaction.rst
Comment (norosa) [inline docs/sql/statements/start-transaction.rst:16]:
```suggestion


Synopsis
========
```

nit: two empty lines before each heading

File: docs/sql/statements/start-transaction.rst
Comment (norosa) [inline docs/sql/statements/start-transaction.rst:22]:
```suggestion
Where ``transaction_mode`` is one of::
```

I think the SQL synopsis sections, in general, could do with a bit of reworking to expand the explanation of what is being documented. however, this probably _isn't_ the PR to start rethinking that

added this suggestion in the meantime as a minor, easy thing to improve

File: docs/sql/statements/start-transaction.rst
Comment (norosa) [inline docs/sql/statements/start-transaction.rst:26]:
```suggestion
Where ``isolation_level`` is one of::
```

File: docs/sql/statements/start-transaction.rst
Comment (norosa) [inline docs/sql/statements/start-transaction.rst:31]:
```suggestion


Description
===========
```

File: docs/sql/statements/start-transaction.rst
Comment (norosa) [inline docs/sql/statements/start-transaction.rst:36]:
```suggestion
CrateDB does not support transactions and will silently ignore the ``START
TRANSACTION`` statement.
```

something like this maybe? I'm cautious about documenting the _intended_ function of the statement when it is a no-op for CrateDB users. my guess is that it would be more confusing than it is useful. but again, happy to take input on this as I probably do not have the full picture

File: docs/sql/statements/start-transaction.rst
Comment (norosa) [inline docs/sql/statements/start-transaction.rst:42]:
```suggestion
  This parameter has no effect.
```

something terse like this maybe? for the same reasons as above

File: docs/interfaces/postgres.rst
Comment (norosa) [inline docs/interfaces/postgres.rst:271]:
```suggestion
CrateDB will silently ignore the ``COMMIT TRANSACTION``, ``BEGIN TRANSACTION``,
and ``START TRANSACTION statements and all respective parameters.
```

reworded this to use the [active voice](https://www.grammarly.com/blog/active-vs-passive-voice/). I also expanded the statement literals for precision and clarity

I removed the "since" part because I think its redundant given the rest of this subsection

**Review by norosa — COMMENTED** [review]
thanks for the PR!

I left some comments. they are only suggestions so I am leaving this as a comment review only

the issue I see here is that the existing pages (including the new ones relating to transactions) could do with some improvements. however, I understand why new pages would copy existing ones (especially when they're closely related)

I am not sure what makes the most sense:

- try to improve things in a single isolated PR like this
- copy the style/structure _as is_ and come back to these pages afterward and make the same improvements to all of them at once

I will leave it up to you and the other reviewers to decide which improvements should go in and which we should hold off on. I would be totally happy with just the very minor fixes I suggested going in and us waiting (I am also totally happy to pick up the work afterward and make the changes myself as a sort of "second pass")

**Comment by jeeminso** [comment]
Thanks for your feedback I will take the first approach and merge all your suggestions.

File: docs/sql/statements/start-transaction.rst
Comment (jeeminso) [reply docs/sql/statements/start-transaction.rst:8]: Yes I agree. Aligned with your idea where you suggested to remove `The statement starts a transaction block until it committed or rolled back.` from the description, I think we should not mention `starts a transaction block` in the intro. Technically it is no-op and may confuse people. 
And also as you said, instead of explaining how to use a statement we do not support, we should mention something like "This is PostgreSQL compatibility feature that the statement will be accept but will be silently ignored."

File: docs/sql/statements/start-transaction.rst
Comment (jeeminso) [reply docs/sql/statements/start-transaction.rst]: how do I reference this new syntax? `.. sql-start-transaction:`  I tried :ref:`START <ref-sql-start-transaction>` but does not work ..

Comment (norosa) [reply docs/sql/statements/start-transaction.rst]: the entire label is opaque (from a syntax perspective) so the `ref` prefix doesn't mean anything special. to reference a label named `sql-start-transaction` you would use this syntax:

    :ref:`example <sql-start-transaction>`

Comment (norosa) [reply docs/sql/statements/start-transaction.rst]: oops my bad I just noticed I accidentally missed the initial `_` character

my original suggested change was:

    .. sql-start-transaction:

my corrected suggestion is:

    .. _sql-start-transaction:

File: docs/appendices/release-notes/unreleased.rst
Comment (norosa) [inline docs/appendices/release-notes/unreleased.rst:76]:
```suggestion
- CrateDB now accepts the ``START TRANSACTION`` statement for :ref:`PostgreSQL
  wire protocol <postgres_wire_protocol>` compatibility. However, CrateDB does
  not support transactions and will silently ignore this statement.
```

changes:

- reworded it slightly so we say CrateDB merely "accepts" the statement (as opposed to "supports") because "supports" may imply something more than a no-op. additionally, we use the word "support" later in this bullet point to say CrateDB _doesn't_ support transactions. reserving "support" for "this feature is implemented and is functional" probably improves clarity/precision
- expanded "Postgres" to "PostgreSQL" because we should use product names in full
- expanded "PostgresSQL" to "PostgreSQL wire protocol" and added a link to the respective page in the documentation (its usually good to cross-reference other parts of the docs when possible, even from the release notes)
- chopped the sentence into two to simplify reading. reworded the second sentence to make CrateDB more of an active participant

File: docs/interfaces/postgres.rst
Comment (norosa) [inline docs/interfaces/postgres.rst:260]: how about:

    For compatibility with clients that use the PostgresSQL wire protocol (e.g.,
    the Golang lib/pq and pgx drivers), CrateDB will accept the :ref:`BEGIN
    <ref-begin>`, :ref:`COMMIT <ref-commit>`, and :ref:`START TRASNACTION
    <sql-start-transaction>` statements. For example::

changes here:

- fixed "Postgres" (as above)
- tidied up the grammar/phrasing (I realize this is the text you are modifying, but while we're in here we may as well tidy it up a little) :)
- expanded ``START`` to ``START TRANSACTION`` so we're using the statement in full (better to be as explicit as possible)
- reordered the list of statements to match the list of statements that appear later down this sectiobn
- I reworded things slightly so that we say that CrateDB "accepts" the statements. I think "implements" and "supports" may confuse matters slightly because both of them _may_ give the impression that these statements are something more than a no-op

File: docs/sql/statements/start-transaction.rst
Comment (norosa) [inline docs/sql/statements/start-transaction.rst:10]:
```suggestion
CrateDB accepts the ``START TRANSACTION` statement for compatibility with the
:ref:`PostgreSQL wire protocol <postgres_wire_protocol>`. However, CrateDB does
not support transactions and will silently ignore this statement.

.. SEEALSO::

    :ref:`Appendix: SQL compatibility <crate_standard_sql>`
```

cool! thanks for updating this bit

I took a second look at this file with your new changes and I think we could improve it by moving the bulk of the info to this para

my suggested change:

- reword a little so CrateDB is more of an active participant (i.e., the noun doing the verbs)
- made `START TRANSACTION` explicit in the first instance (and "this statement" in the second instance)
- expanded and link "PostgreSQL wire protocol"
- moved up your link to `crate_standard_sql` from below (good addition!) and turned it into a `SEEALSO` block with explicit link text. because this is a `SEEALSO` block, I added an "Appendix: " prefix to identify which section of the docs this page belongs to

File: docs/sql/statements/start-transaction.rst
Comment (norosa) [inline docs/sql/statements/start-transaction.rst:40]:
```suggestion
CrateDB will silently ignore the ``START TRANSACTION`` statement.
```

because we moved this info to the top where readers will see it first, we can afford to trim this bit down by removing any explanation or justification

**Comment by jeeminso** [comment]
@norosa Thanks Naomi for fixing the docs and providing the reasons behind it, will try to refer to this PR when I would touch on docs again.

**Pull request merged**
